SplitTests/SplitFilter/FilterBuilderTest.swift

```diff
@@ -54,6 +80,20 @@ class FilterBuilderTest: XCTestCase {
         XCTAssertEqual("&names=nf_a,nf_b,nf_c,nf_d&prefixes=pf_a,pf_b,pf_c,pf_d", queryString);
     }
 
+    func testFilterBySetsValuesDeduptedAndGrouped() throws {
+        // Duplicated filter values should be removed on builing
+        let filters: [SplitFilter] = [
+                        SplitFilter.bySet(["nf_a", "nf_c", "nf_b"]),
+                        SplitFilter.bySet(["nf_b", "nf_d"]),
+                        SplitFilter.byPrefix(["pf_a", "pf_c", "pf_b"]),
+                        SplitFilter.byPrefix(["pf_d", "pf_a"])
+        ]
+
+        let queryString = try FilterBuilder().add(filters: filters).build()
+
+        XCTAssertEqual("&set=nf_a,nf_b,nf_c,nf_d&prefixes=pf_a,pf_b,pf_c,pf_d", queryString);
```

Comment:
Prefixes should be discarded since there are bySet filters.
```suggestion
        XCTAssertEqual("&set=nf_a,nf_b,nf_c,nf_d", queryString);
```